packages/tests/backend/sirius-components-graphql-tests/src/main/java/org/eclipse/sirius/components/graphql/tests/ExecuteEditingContextFunctionEventHandler.java

```diff
@@ -35,10 +35,18 @@ public boolean canHandle(IEditingContext editingContext, IInput input) {
     }
 
     @Override
+    @SuppressWarnings("checkstyle:IllegalCatch")
     public void handle(Sinks.One<IPayload> payloadSink, Sinks.Many<ChangeDescription> changeDescriptionSink, IEditingContext editingContext, IInput input) {
         if (input instanceof ExecuteEditingContextFunctionInput executeEditingContextFunctionInput) {
-            var result = executeEditingContextFunctionInput.function().apply(editingContext);
-            payloadSink.tryEmitValue(new ExecuteEditingContextFunctionSuccessPayload(input.id(), result));
+            Object result = null;
+            try {
+                result = executeEditingContextFunctionInput.function().apply(editingContext);
+            } catch (Throwable e) {
+                payloadSink.tryEmitValue(new ErrorPayload(input.id(), e.getMessage()));
+            }
+            if (result != null) {
+                payloadSink.tryEmitValue(new ExecuteEditingContextFunctionSuccessPayload(input.id(), result));
+            }
```

Comment:
Wouldn't that send two ErrorPayload in case of an exception?

Reply:
Interestingly this wasn't an issue although the existing code was already producing multiple payloads if the function call succeeded (the success payload and the error payload after the `if`). I fixed it to use the pattern we have in other handlers with a single `tryEmitValue` call.